From 5e9813ea9c6794307937af2c8bf709b99e065468 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Wed, 17 May 2017 14:46:46 -0400
Subject: [PATCH 1/2] block: remove unused block format support

block.c need to be modified to remove references to the disabled qcow2
format.
---
 block.c             |  6 ++++--
 block/Makefile.objs | 15 ++++++++-------
 2 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/block.c b/block.c
index 951f5f4..3bad8ef 100644
--- a/block.c
+++ b/block.c
@@ -1405,7 +1405,7 @@ done:
     qdict_del(options, bdref_key);
     return c;
 }
-
+#if 0
 static int bdrv_append_temp_snapshot(BlockDriverState *bs, int flags,
                                      QDict *snapshot_options, Error **errp)
 {
@@ -1471,7 +1471,7 @@ out:
     g_free(tmp_filename);
     return ret;
 }
-
+#endif
 /*
  * Opens a disk image (raw, qcow2, vmdk, ...)
  *
@@ -1702,8 +1702,10 @@ static int bdrv_open_inherit(BlockDriverState **pbs, const char *filename,
     /* For snapshot=on, create a temporary qcow2 overlay. bs points to the
      * temporary snapshot afterwards. */
     if (snapshot_flags) {
+#if 0
         ret = bdrv_append_temp_snapshot(bs, snapshot_flags, snapshot_options,
                                         &local_err);
+#endif
         snapshot_options = NULL;
         if (local_err) {
             goto close_and_fail;
diff --git a/block/Makefile.objs b/block/Makefile.objs
index cdb33cc..ab80617 100644
--- a/block/Makefile.objs
+++ b/block/Makefile.objs
@@ -1,10 +1,11 @@
-block-obj-y += raw_bsd.o qcow.o vdi.o vmdk.o cloop.o bochs.o vpc.o vvfat.o
-block-obj-y += qcow2.o qcow2-refcount.o qcow2-cluster.o qcow2-snapshot.o qcow2-cache.o
-block-obj-y += qed.o qed-gencb.o qed-l2-cache.o qed-table.o qed-cluster.o
-block-obj-y += qed-check.o
+block-obj-y += raw_bsd.o
+block-obj-n += qcow.o vdi.o vmdk.o cloop.o bochs.o vpc.o vvfat.o
+block-obj-n += qcow2.o qcow2-refcount.o qcow2-cluster.o qcow2-snapshot.o qcow2-cache.o
+block-obj-n += qed.o qed-gencb.o qed-l2-cache.o qed-table.o qed-cluster.o
+block-obj-n += qed-check.o
 block-obj-$(CONFIG_VHDX) += vhdx.o vhdx-endian.o vhdx-log.o
-block-obj-y += quorum.o
-block-obj-y += parallels.o blkdebug.o blkverify.o blkreplay.o
+block-obj-n += quorum.o
+block-obj-n += parallels.o blkdebug.o blkverify.o blkreplay.o
 block-obj-y += block-backend.o snapshot.o qapi.o
 block-obj-$(CONFIG_WIN32) += raw-win32.o win32-aio.o
 block-obj-$(CONFIG_POSIX) += raw-posix.o
@@ -13,7 +14,7 @@ block-obj-$(CONFIG_ATAPI_PT) += atapi-pt-protocol.o
 block-obj-y += null.o mirror.o io.o
 block-obj-y += throttle-groups.o
 
-block-obj-y += nbd.o nbd-client.o sheepdog.o
+block-obj-n += nbd.o nbd-client.o sheepdog.o
 block-obj-$(CONFIG_LIBISCSI) += iscsi.o
 block-obj-$(CONFIG_LIBNFS) += nfs.o
 block-obj-$(CONFIG_CURL) += curl.o
-- 
2.9.4

